src/lightgbm/src/main/scala/LightGBMBase.scala

```diff
@@ -64,15 +64,26 @@ trait LightGBMBase[TrainedModel <: Model[TrainedModel]] extends Estimator[Traine
       else None
     val lightGBMBooster = df
       .mapPartitions(TrainUtils.trainLightGBM(networkParams, getLabelCol, getFeaturesCol, get(weightCol),
-        validationData, log, trainParams, numCoresPerExec))(encoder)
+        getOptGroupCol, validationData, log, trainParams, numCoresPerExec))(encoder)
       .reduce((booster1, _) => booster1)
     // Wait for future to complete (should be done by now)
     Await.result(future, Duration(getTimeout, SECONDS))
     getModel(trainParams, lightGBMBooster)
   }
 
+  /** Optional group column for Ranking, set to None by default.
+    * @return None
+    */
+  protected def getOptGroupCol: Option[String] = None
```

Comment:
WHat is this func for?maybe you are looking for `get(groupCol)`?

Reply:
it's just the group column that I pass to map partitions and is required for lightgbm ranker but not needed for classifier or regressor